crates/sui-indexer/src/main.rs

```diff
@@ -46,12 +46,22 @@ async fn main() -> anyhow::Result<()> {
             snapshot_config,
             pruning_options,
             upload_options,
+            mvr_mode,
         } => {
             // Make sure to run all migrations on startup, and also serve as a compatibility check.
             run_migrations(pool.dedicated_connection().await?).await?;
-            let retention_config = pruning_options.load_from_file();
+            let mut retention_config = pruning_options.load_from_file();
             if retention_config.is_some() {
                 check_prunable_tables_valid(&mut pool.get().await?).await?;
+            } else {
```

Comment:
the logic here is a bit strange, in case of `mvr_mode`, we want a custom RetentionConfig, otherwise indexer is in vanilla mode and `load_from_file`, following this the codes should be sth like?
```
let retention_config = if mvr_mode { /* custom value */ }  else {pruning_options.load_from_file()};
if retention_config.is_some() {
  check_prunable_tables_valid(&mut pool.get().await?).await?;
}
```
vs now it loads config anyways but likely discarded right away?

Reply:
yes that's cleaner, thank you for the suggestion